CONTRIBUTING.rst

```diff
@@ -0,0 +1,114 @@
+Contributing
+============
+
+Contributions are welcomed via `pull requests on
+GitHub <https://github.com/glotzerlab/hoomd-blue/pulls>`__. Contact the
+**HOOMD-blue** developers before starting work to ensure it meshes well
+with the planned development direction and standards set for the
+project.
+
+Features
+--------
+
+Implement functionality in a general and flexible fashion
+_________________________________________________________
+
+New features should be applicable to a variety of use-cases. The
+**HOOMD-blue** developers can assist you in designing flexible
+interfaces.
+
+Maintain performance of existing code paths
+___________________________________________
+
+Expensive code paths should only execute when requested.
+
+Optimize for the current GPU generation
+_______________________________________
+
+Write, test, and optimize your GPU kernels on the latest generation of
+GPUs.
+
+Version control
+---------------
+
+Base your work off the correct branch
+_____________________________________
+
+During the v3.0.0-beta release cycle, all work must be based on
+``master``.
+
+Propose a minimal set of related changes
+________________________________________
+
+All changes in a pull request should be closely related. Multiple change
+sets that are loosely coupled should be proposed in separate pull
+requests.
+
+Agree to the Contributor Agreement
+__________________________________
+
+All contributors must agree to the Contributor Agreement before their
+pull request can be merged.
+
+Source code
+-----------
+
+Use a consistent style
+______________________
+
+The **Code style** section of the documentation sets the style
+guidelines for **HOOMD-blue** code.
+
+Document code with comments
+___________________________
+
+Use doxygen header comments for classes, functions, etc. Also comment
+complex sections of code so that other developers can understand them.
+
+Compile without warnings
+________________________
+
+Your changes should compile without warnings.
+
+Tests
+-----
+
+Write unit tests
+________________
+
+Add unit tests for all new functionality.
+
+Validity tests
+______________
+
+The developer should run research-scale simulations using the new
+functionality and ensure that it behaves as intended.
+
+User documentation
+------------------
+
+Write user documentation
+________________________
+
+Document public-facing API with Python docstrings in Google style.
+
+Document version status
+_______________________
+
+Each user-facing Python class, method, etc. with a docstring should have
+`versionadded, versionchanged, and deprecated Sphinx
+directives <https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded>`__
+so that users will be aware of how functionality changes from version to
+version.
```

Comment:
Have we been following this? If this is expected then we need to add version 3.0 to a lot of docstrings if not remove.

Reply:
I will update this. The unwritten guideline we have been following is that everything is implicitly new in the most recent major release and all `versionadded` and `versionchanged` directives should be removed in 3.0.0. We will start following this with new/changed features in 3.1.